syft/frameworks/torch/tensors/interpreters/chinese_remainder_theorem.py

```diff
@@ -0,0 +1,394 @@
+import torch
+import math
+import itertools
+
+import syft
+from syft.frameworks.torch.tensors.interpreters.abstract import AbstractTensor
+from syft.frameworks.torch.tensors.interpreters.precision import FixedPrecisionTensor
+from syft.frameworks.torch.tensors.interpreters.additive_shared import AdditiveSharingTensor
+from syft.frameworks.torch.overload_torch import overloaded
+
+
+class CRTTensor(AbstractTensor):
+    """ A CRT tensor is a tensor whose values are represented as their remainders modulo several pairwise coprime numbers.
+    The true tensor values lie in the Zq field where q is the product of all the moduli.
+    A CRT tensor then represent a modular system of equations:
+        x = a_0 mod f_0
+        ...
+        x = a_n mod f_n
+    The Chineses Remainder Theorem (that also gives his name to this tensor) assert that exactly one x in Zq with q = f_0 * ... * f_n
+    satisfies this system. This x is the real value represented by the tensor.
+    This tensor makes it possible to represent big numbers and to avoid overflows while manipulating them.
+    It also makes additions, subtractions, and multiplications of huge numbers quite efficient.
+    """
+
+    def __init__(
+        self,
+        field: int = 21,
+        residues: dict = None,
+        base=None,
+        precision_fractional=None,
+        reconstruction_coeffs=None,
+        owner=None,
+        id=None,
+        tags: set = None,
+        description: str = None,
+    ):
+        super().__init__(owner=owner, id=id, tags=tags, description=description)
+
+        # Check that all the moduli are pairwise coprime
+        if residues is not None:
+            for pair in itertools.combinations(residues.keys(), r=2):
```

Comment:
Can we add a flag to skip this test, if the list of keys gets long, this test takes a long time and if we could verify it offline, that would a ton of time (also after the initial spin up) (or if we could cache the results of the function)

Reply:
This test will occur only when the user gives residues, not when the "usual" way of converting to CRT tensors is used (by calling .fix_prec(storage="crt"). So this should not be that often...
But I can still only warn the user that the check is not done if the list is too long. What would you consider a long list in this case?